Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/MainActivity.java

```diff
@@ -68,16 +70,8 @@ protected void onCreate(Bundle savedInstanceState)
 
     if (PermissionsHandler.hasWriteAccess(this))
     {
-      PlatformPagerAdapter platformPagerAdapter = new PlatformPagerAdapter(
-              getSupportFragmentManager(), this);
-      mViewPager.setAdapter(platformPagerAdapter);
-      mViewPager.setOffscreenPageLimit(platformPagerAdapter.getCount());
-      showGames();
-      GameFileCacheService.startLoad(this);
-    }
-    else
-    {
-      mViewPager.setVisibility(View.INVISIBLE);
```

Comment:
Was removing this intentional?

Reply:
Yes, the original code would call `mTabLayout.setupWithViewPager(mViewPager)` before write permission was obtained and change visibility accordingly. It seemed simpler to call mTabLayout.setupWithViewPager(mViewPager) after write permission was obtained, which removes the need to change visibility.